### PR TITLE
updating models

### PR DESCRIPTION
**Pull request by fivetran-sheringuyen**
**Are you a current Fivetran customer?** 

Fivetran employee

**What change(s) does this PR introduce?** 

## Features
- Soft deleted records are now included in the below models/macros:
  - `macros/email_events_joined`
  - `models/marketing/intermediate/int_hubspot__email_metrics__by_contact_list`
  - `models/sales/intermediate/int_hubspot__deals_enhanced`
  - `models/sales/deal_stages`

- Soft deleted records can now be removed by using the following respective flags for the above models:
  - `is_contact_list_member_deleted`
  - `is_deal_pipeline_deleted`
  - `is_deal_pipeline_stage_deleted`
  - `is_deal_deleted`

**Did you update the CHANGELOG?** 

- [x] Yes

**Does this PR introduce a breaking change?**

- [ ] Yes (please provide breaking change details below.)
- [x] No  (please provide an explanation as to how the change is non-breaking below.)

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [x] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [x] Yes, Issue/Feature [dbt hubspot source#87](https://github.com/fivetran/dbt_hubspot_source/issues/87)
- [ ] No 

**How did you test the PR changes?** 

 

- [x] Buildkite  
- [x] Local (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 

- [x] BigQuery
- [x] Redshift
- [x] Snowflake
- [x] Postgres
- [x] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
:dancer:

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

File: models/sales/sales.yml
Comment (fivetran-jamie) [inline models/sales/sales.yml:152]: we also need to replace `is_deleted` and `_fivetran_deleted` with `is_<table name>_deleted` in the other models (ie contact_lists, contacts, comapnies etc)

File: models/marketing/intermediate/int_hubspot__email_metrics__by_contact_list.sql
Comment (fivetran-jamie) [inline models/marketing/intermediate/int_hubspot__email_metrics__by_contact_list.sql:18]: what's the purpose of bringing in the is_contact_list_member_deleted field here, since it's in aggregational model? do we need to bring it into the contact_list end model or sum up the email_metrics again there to avoid fanout? i am getting some errors when running `dbt test`

```
Completed with 11 errors and 0 warnings:
21:10:22  
21:10:22  Failure in test unique_hubspot__contact_lists_contact_list_id (models/marketing/marketing.yml)
21:10:22    Got 7 results, configured to fail if != 0
21:10:22  
21:10:22    compiled Code at target/compiled/hubspot/models/marketing/marketing.yml/unique_hubspot__contact_lists_contact_list_id.sql
21:10:22  
21:10:22  Failure in test unique_hubspot__email_event_bounce_event_id (models/marketing/email_events/email_events.yml)
21:10:22    Got 14 results, configured to fail if != 0
21:10:22  
21:10:22    compiled Code at target/compiled/hubspot/models/marketing/email_events/email_events.yml/unique_hubspot__email_event_bounce_event_id.sql
21:10:22  
21:10:22  Failure in test unique_hubspot__email_event_clicks_event_id (models/marketing/email_events/email_events.yml)
21:10:22    Got 7 results, configured to fail if != 0
21:10:22  
21:10:22    compiled Code at target/compiled/hubspot/models/marketing/email_events/email_events.yml/unique_hubspot__email_event_clicks_event_id.sql
21:10:22  
21:10:22  Failure in test unique_hubspot__email_event_deferred_event_id (models/marketing/email_events/email_events.yml)
21:10:22    Got 4 results, configured to fail if != 0
21:10:22  
21:10:22    compiled Code at target/compiled/hubspot/models/marketing/email_events/email_events.yml/unique_hubspot__email_event_deferred_event_id.sql
21:10:22  
21:10:22  Failure in test unique_hubspot__email_event_delivered_event_id (models/marketing/email_events/email_events.yml)
21:10:22    Got 336 results, configured to fail if != 0
21:10:22  
21:10:22    compiled Code at target/compiled/hubspot/models/marketing/email_events/email_events.yml/unique_hubspot__email_event_delivered_event_id.sql
21:10:22  
21:10:22  Failure in test unique_hubspot__email_event_dropped_event_id (models/marketing/email_events/email_events.yml)
21:10:22    Got 42 results, configured to fail if != 0
21:10:22  
21:10:22    compiled Code at target/compiled/hubspot/models/marketing/email_events/email_events.yml/unique_hubspot__email_event_dropped_event_id.sql
21:10:22  
21:10:22  Failure in test unique_hubspot__email_event_opens_event_id (models/marketing/email_events/email_events.yml)
21:10:22    Got 1218 results, configured to fail if != 0
21:10:22  
21:10:22    compiled Code at target/compiled/hubspot/models/marketing/email_events/email_events.yml/unique_hubspot__email_event_opens_event_id.sql
21:10:22  
21:10:22  Failure in test unique_hubspot__email_event_status_change_event_id (models/marketing/email_events/email_events.yml)
21:10:22    Got 2 results, configured to fail if != 0
21:10:22  
21:10:22    compiled Code at target/compiled/hubspot/models/marketing/email_events/email_events.yml/unique_hubspot__email_event_status_change_event_id.sql
21:10:22  
21:10:22  Failure in test unique_hubspot__email_event_sent_event_id (models/marketing/email_events/email_events.yml)
21:10:22    Got 724 results, configured to fail if != 0
21:10:22  
21:10:22    compiled Code at target/compiled/hubspot/models/marketing/email_events/email_events.yml/unique_hubspot__email_event_sent_event_id.sql
21:10:22  
21:10:22  Failure in test unique_hubspot__email_sends_event_id (models/marketing/marketing.yml)
21:10:22    Got 724 results, configured to fail if != 0
21:10:22  
21:10:22    compiled Code at target/compiled/hubspot/models/marketing/marketing.yml/unique_hubspot__email_sends_event_id.sql
21:10:22  
21:10:22  Failure in test unique_int_hubspot__email_metrics__by_contact_list_contact_list_id (models/marketing/intermediate/intermediate.yml)
21:10:22    Got 7 results, configured to fail if != 0
```

**Review by fivetran-jamie — CHANGES_REQUESTED** [review]
left some in-line comments! mostly docs-focused except the int_hubspot__email_metrics_by_contact_list

File: CHANGELOG.md
Comment (fivetran-sheringuyen) [reply CHANGELOG.md:3]: yes! Forgot to add to the changelog after creating the PR, thank you for the reminder!

File: CHANGELOG.md
Comment (fivetran-sheringuyen) [reply CHANGELOG.md:1]: Will be making this a breaking change per discussion in source pkg

File: CHANGELOG.md
Comment (fivetran-sheringuyen) [reply CHANGELOG.md:8]: oh snap I missed this, great catch!

File: CHANGELOG.md
Comment (fivetran-sheringuyen) [reply CHANGELOG.md:7]: Great idea! I will add another section for this.

File: models/marketing/intermediate/int_hubspot__email_metrics__by_contact_list.sql
Comment (fivetran-sheringuyen) [reply models/marketing/intermediate/int_hubspot__email_metrics__by_contact_list.sql:18]: hm I think you're right here! I think when I started I just followed along with [the plan](https://github.com/fivetran/dbt_hubspot_source/issues/87#issuecomment-1315674869) but I agree, I don't think it's actually necessary to include it here.

File: packages.yml
Comment (fivetran-jamie) [inline packages.yml:4]: obligatory reminder to update before merging

File: models/marketing/marketing.yml
Comment (fivetran-jamie) [inline models/marketing/marketing.yml:326]: we need to add `is_contact_list_deleted` right?

File: models/marketing/marketing.yml
Comment (fivetran-jamie) [inline models/marketing/marketing.yml:209]: i think `hubspot__email_sends` is missing this

File: CHANGELOG.md
Comment (fivetran-jamie) [inline CHANGELOG.md:7]: `hubspot__email_sends` as well i think

**Review by fivetran-jamie — COMMENTED** [review]
looks pretty good -- found a couple of fields missing docs and still getting failures on uniqueness tests on email event models tho

**Review by fivetran-jamie — APPROVED** [review]
approving now that i see the dbt test failures on main as well... that's another issue we should investigate outside of this PR. so once docs are updated lgtm

**Pull request merged**
